httpsupport/http_client_blackbox_test.go

```diff
@@ -0,0 +1,94 @@
+package httpsupport_test
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-common/httpsupport"
+	"github.com/fabric8-services/fabric8-common/resource"
+
+	"github.com/goadesign/goa"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHost(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	req := &goa.RequestData{
+		Request: &http.Request{Host: "api.service.domain.org"},
+	}
+
+	t.Run("prod mode", func(t *testing.T) {
+		host := httpsupport.Host(req, &dummyConfig{})
+		assert.Equal(t, "api.service.domain.org", host)
+	})
+
+	t.Run("nil config", func(t *testing.T) {
+		host := httpsupport.Host(req, nil)
+		assert.Equal(t, "api.service.domain.org", host)
+	})
+
+	t.Run("dev mode", func(t *testing.T) {
+		host := httpsupport.Host(req, &dummyConfig{true})
+		assert.Equal(t, "auth.openshift.io", host)
+	})
+}
+
+func TestAddParams(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	t.Run("success", func(t *testing.T) {
+		testMap := map[string]string{
+			"param1": "a",
+			"param2": "b",
+			"param3": "https://www.redhat.com",
+		}
+		testHost := "openshift.io"
+		generatedURLString, err := httpsupport.AddParams("https://"+testHost, testMap)
+		require.NoError(t, err)
+
+		generateURL, err := url.Parse(generatedURLString)
+		require.NoError(t, err)
+
+		assert.Equal(t, testHost, generateURL.Host)
+		assert.Equal(t, "https", generateURL.Scheme)
+
+		m, _ := url.ParseQuery(generateURL.RawQuery)
+		for k, v := range testMap {
+			assert.Equal(t, v, m[k][0])
+		}
+	})
+}
+
+func TestAddParamSuccess(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	t.Parallel()
+
+	t.Run("success", func(t *testing.T) {
+		generatedURL, err := httpsupport.AddParam("https://openshift.io", "param1", "a")
+		require.NoError(t, err)
+		assert.Equal(t, "https://openshift.io?param1=a", generatedURL)
+
+		generatedURL, err = httpsupport.AddParam(generatedURL, "param2", "abc")
+		require.NoError(t, err)
+		equalURLs(t, "https://openshift.io?param1=a&param2=abc", generatedURL)
+	})
```

Comment:
this looks nice test :-) but I guess better to separate both test.  For ex, `generatedURL` from first test itself is wrong then passing wrong thing to test2 as input would not be good.
Also, for test1, I guess you forgot to use `equalURLs()`.

Reply:
good point, let me refactor those 2 tests into something more unified.

Reply:
done in aeb26b3